src/audio/kpb.c

```diff
@@ -114,6 +119,33 @@ static uint64_t kpb_task_deadline(void *data)
 	return SOF_TASK_DEADLINE_ALMOST_IDLE;
 }
 
+static void kpb_lock(struct comp_data *kpb)
+{
+#ifdef __ZEPHYR__
+	k_mutex_lock(&kpb->lock, K_FOREVER);
+#else
+	kpb->key = k_spin_lock(&kpb->lock);
+#endif
+}
+
+static void kpb_unlock(struct comp_data *kpb)
+{
+#ifdef __ZEPHYR__
+	k_mutex_unlock(&kpb->lock);
+#else
+	k_spin_unlock(&kpb->lock, kpb->key);
+#endif
+}
+
+static void kpb_lock_init(struct comp_data *kpb)
+{
+#ifdef __ZEPHYR__
+	k_mutex_init(&kpb->lock);
+#else
+	k_spinlock_init(&kpb->lock);
+#endif
```

Comment:
Far nicer to have a single conditional ifdef for these functions i.e.
```c
#if ZEPHYR 
//all func here using mutex
#else
//all funcs here using spinlock
#endif 
```

Reply:
There's also `#if CONFIG_INCOHERENT` there, so, you anyway end up with multi-level conditionals. I can try to invert - move `__ZEPHYR__` outside, not sure it will be better. And the diff would probably be larger

Reply:
CONFIG_COHERENT should not be visible to kpb.c, only the ZEPHYR part is appropriate.

Reply:
sorry, I misunderstood which commit this comment was referring to